lib/mayaUsd/utils/util.h

```diff
@@ -66,7 +66,9 @@ struct _CmpDag
 {
     bool operator()(const MDagPath& lhs, const MDagPath& rhs) const
     {
-        return strcmp(lhs.fullPathName().asChar(), rhs.fullPathName().asChar()) < 0;
+        int pathCountDiff = lhs.pathCount() - rhs.pathCount();
+        return (0 != pathCountDiff) ? (pathCountDiff < 0) :
+                (strcmp(lhs.fullPathName().asChar(), rhs.fullPathName().asChar()) < 0);
```

Comment:
Changing the sorting function to more explicitly sort by hierarchy depth, and then alphabetically within the same depth.  End result is the same because of the path separator used in the fullPathName, but adding the pathCount check makes it more obvious that it’s doing this.  It would also be faster this way since the path count is a stored value internally so it’s a quick int compare in a lot of cases, avoiding a bunch of string compares.